pkg/utils/utils.go

```diff
@@ -70,3 +72,35 @@ func DecompressGZIP(compressedData []byte) (decompressedData []byte, err error)
 
 	return decompressedData, nil
 }
+
+// TODO: check if it's worth trying to compress the data prior to compressing it (e.g: using entropy algorithms)
+// This would save some CPU usage client side
+// TODO: have telemetry data type be passed in as a parameter to further check if we should compress data and which algorithm to use (e.g: deflate or gzip)
+func ShouldCompress(data []byte) (resultData []byte, compression *string, err error) {
```

Comment:
The name shouldCompress() suggests a check that returns a boolean indicating whether the data should be compressed or not. However this routine is actually returning the data potentially compressed, and the compression method used.

So a better name might be something like compressionFilter() or compressIfNeeded()